test/kubernetes/e2e/features/services/tcproute/suite.go

```diff
@@ -31,95 +37,271 @@ func NewTestingSuite(ctx context.Context, testInst *e2e.TestInstallation) suite.
+			}
+
+			// Assert TCPRoute conditions
+			for _, tcpRouteName := range tc.tcpRouteNames {
+				s.testInstallation.Assertions.EventuallyTCPRouteCondition(s.ctx, tcpRouteName, tc.gtwNs, v1.RouteConditionAccepted, metav1.ConditionTrue, timeout)
+				s.testInstallation.Assertions.EventuallyTCPRouteCondition(s.ctx, tcpRouteName, tc.gtwNs, v1.RouteConditionResolvedRefs, expected, timeout)
+			}
+
+			// Assert gateway programmed condition
+			s.testInstallation.Assertions.EventuallyGatewayCondition(s.ctx, tc.gtwName, tc.gtwNs, v1.GatewayConditionProgrammed, metav1.ConditionTrue, timeout)
+
+			// Assert listener attached routes
+			for i, listenerName := range tc.listenerNames {
+				expectedRouteCount := tc.expectedRouteCounts[i]
+				s.testInstallation.Assertions.EventuallyGatewayListenerAttachedRoutes(s.ctx, tc.gtwName, tc.gtwNs, listenerName, expectedRouteCount, timeout)
+			}
+
+			// Assert expected responses
+			for i, port := range tc.ports {
+				if tc.name == "CrossNamespaceTCPRouteWithoutReferenceGrant" {
+					s.testInstallation.Assertions.AssertEventualCurlError(
+						s.ctx,
+						s.execOpts(tc.gtwNs),
+						[]curl.Option{
+							curl.WithHost(kubeutils.ServiceFQDN(tc.proxyService.ObjectMeta)),
+							curl.WithPort(port),
+							curl.VerboseOutput(),
+						},
+						tc.expectedErrorCode)
+				} else {
+					s.testInstallation.Assertions.AssertEventualCurlResponse(
+						s.ctx,
+						s.execOpts(tc.gtwNs),
+						[]curl.Option{
+							curl.WithHost(kubeutils.ServiceFQDN(tc.proxyService.ObjectMeta)),
+							curl.WithPort(port),
+							curl.VerboseOutput(),
+						},
+						tc.expectedResponses[i])
+				}
+			}
+		})
+	}
+}
+
+func validateManifestFile(path string) error {
+	if _, err := os.Stat(path); os.IsNotExist(err) {
+		return fmt.Errorf("Manifest file not found: %s", path)
+	}
+	return nil
+}
+
+func (s *testingSuite) setupTestEnvironment(nsManifest, gtwName, gtwNs, gtwManifest, svcManifest string, proxySvc *corev1.Service, proxyDeploy *appsv1.Deployment) {
+	s.applyManifests(nsManifest)
+
+	s.applyManifests(gtwManifest)
+	s.testInstallation.Assertions.EventuallyGatewayCondition(s.ctx, gtwName, gtwNs, v1.GatewayConditionAccepted, metav1.ConditionTrue, timeout)
+
+	s.applyManifests(svcManifest)
+	s.testInstallation.Assertions.EventuallyObjectsExist(s.ctx, proxySvc, proxyDeploy)
+}
+
+func (s *testingSuite) applyManifests(manifests ...string) {
+	for _, manifest := range manifests {
+		s.Eventually(func() bool {
+			err := s.testInstallation.Actions.Kubectl().ApplyFile(s.ctx, manifest)
+			if err != nil {
+				s.T().Logf("Retrying apply manifest: %s, error: %v", manifest, err)
+				return false
+			}
+			return true
+		}, waitTime, tickTime, fmt.Sprintf("Can apply manifest %s", manifest))
+	}
+}
+
+func (s *testingSuite) deleteManifests(manifests ...string) error {
+	for _, manifest := range manifests {
+		if err := s.testInstallation.Actions.Kubectl().DeleteFile(s.ctx, manifest); err != nil {
```

Comment:
We also have a DeleteFileSafe (https://github.com/solo-io/gloo/blob/main/pkg/utils/kubeutils/kubectl/cli.go#L182)

Reply:
Resolved by commit https://github.com/solo-io/gloo/commit/281ace000966aa7f15d960ab73d60efc4fa1c189 by using `DeleteFileSafe()`.